src/basic/fs-util.c

```diff
@@ -741,18 +741,28 @@ bool unsafe_transition(const struct stat *a, const struct stat *b) {
         return a->st_uid != b->st_uid; /* Otherwise we need to stay within the same UID */
 }
 
-static int log_unsafe_transition(int a, int b, const char *path, unsigned flags) {
+static int log_unsafe_transition(int a, int b, unsigned flags) {
         _cleanup_free_ char *n1 = NULL, *n2 = NULL;
+        _cleanup_free_ char *user_a, *user_b;
```

Comment:
Oh, then these need to be initialized with NULL. And please merge the line with the previous.

Reply:
Done :+1:  Quick question, why is it necessary to initialize with NULL here?

Reply:
Otherwise, the cleanup attribute will free uninitialized address if the function returns earlier before a valid address is assigned to the variable.